src/components/TriggerDeliveryPolicyEditor.tsx

```diff
@@ -0,0 +1,306 @@
+      setIsValidPolicySource(true);
+    } catch (error) {
+      setIsValidPolicySource(false);
+      if (error) {
+        if (error instanceof yup.ValidationError) {
+          setErrorMessage(error.inner[0].message);
+        }
+      }
+    }
+    isSameIntNumbers(policy.error_handlers);
+  };
+
+  const validJSONText = (value: string) => {
+    if (!checkValidJSONText(value)) {
+      setErrorMessage('Invalid JSON!');
+      return;
+    }
+    const newPolicy: AstarteTriggerDeliveryPolicyDTO = JSON.parse(value);
+    isValidJSON(newPolicy);
+  };
+
+  const handlePolicyChange = useCallback((updatedPolicy: AstarteTriggerDeliveryPolicyDTO) => {
+    const isStrategyFieldRetry = updatedPolicy.error_handlers.some((e) => e.strategy === 'retry');
+    const newPolicy: AstarteTriggerDeliveryPolicyDTO = updatedPolicy;
+    if (
+      isStrategyFieldRetry &&
+      (updatedPolicy.retry_times === undefined || updatedPolicy.retry_times === 0)
+    ) {
+      newPolicy.retry_times = 1;
+    }
+    if (!isStrategyFieldRetry) {
+      const { retry_times, ...restElements } = newPolicy;
+      setPolicyDraft(restElements);
+      return;
+    }
+    setPolicyDraft(newPolicy);
+  }, []);
+
+  const handlePolicySourceChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+    setPolicySource(value);
+    validJSONText(value);
+    if (checkValidJSONText(value)) {
+      handlePolicyChange(JSON.parse(value));
+    }
+  };
+
+  useEffect(() => {
+    setPolicySource(JSON.stringify(policyDraft, null, 4));
+    isValidJSON(policyDraft);
+  }, [policyDraft]);
+
+  useEffect(() => {
+    if (onChange) {
+      onChange(policyDraft, isValidPolicySource);
+    }
+  }, [isValidPolicySource, onChange, policyDraft]);
+
+  return (
+    <Row>
+      <Col md={isSourceVisible ? 6 : 12}>
+        <Container fluid className="bg-white rounded p-3">
+          <Form>
+            <Form.Row className="mb-2">
+              <Col sm={12}>
+                <Form.Group controlId="policyName">
+                  <Form.Label>Name</Form.Label>
+                  <Form.Control
+                    type="text"
+                    autoComplete="off"
+                    required
+                    readOnly={isReadOnly}
+                    value={_.get(policyDraft, 'name')}
+                    onChange={handlePolicyNameChange}
+                    isInvalid={!validateName(policyDraft.name)}
+                    spellCheck={false}
+                  />
+                  <Form.Control.Feedback type="invalid">
+                    name is a required field
+                  </Form.Control.Feedback>
+                </Form.Group>
+                <Form.Group controlId="policyHandler">
+                  <TriggerDeliveryPolicyHandler
+                    onChange={handlePolicyChange}
+                    initialData={policyDraft}
+                    isReadOnly={isReadOnly}
+                  />
```

Comment:
I'd suggest to review the behavior of the editor under different scenarios and trying out different combinations for the fields.
For instance, here is a couple of actions that don't work currently and should be fixed:

1. If I create an error handler for custom errors (e.g. 401,403), when I try to edit the error handler the modal does not show the custom errors with the correct values, but instead report `any_error` for the On field. 
2. If I create an error handler with a Retry strategy, when I edit the error handler to change its On field then the policy's source code is not updated.